webapp/src/main/java/com/box/l10n/mojito/CachingConfig.java

```diff
@@ -1,38 +1,119 @@
 package com.box.l10n.mojito;
 
+import com.box.l10n.mojito.service.cache.DatabaseCache;
+import com.box.l10n.mojito.service.cache.DatabaseCachesConfiguration;
+import com.box.l10n.mojito.service.cache.TieredCache;
+import com.box.l10n.mojito.service.machinetranslation.MTServiceCacheConfiguration;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.cache.Cache;
 import org.springframework.cache.CacheManager;
 import org.springframework.cache.annotation.CachingConfigurerSupport;
 import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.cache.caffeine.CaffeineCache;
 import org.springframework.cache.concurrent.ConcurrentMapCache;
 import org.springframework.cache.interceptor.KeyGenerator;
 import org.springframework.cache.support.SimpleCacheManager;
 import org.springframework.context.annotation.AdviceMode;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import java.util.Arrays;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.box.l10n.mojito.service.cache.DatabaseCachesConfiguration.*;
 
 /**
- *
  * @author jaurambault
  */
 @Configuration
 @EnableCaching(mode = AdviceMode.ASPECTJ)
 public class CachingConfig extends CachingConfigurerSupport {
 
+    final MTServiceCacheConfiguration mtServiceCacheConfiguration;
+
+    final DatabaseCachesConfiguration databaseCachesConfiguration;
+
+    public CachingConfig(MTServiceCacheConfiguration mtServiceCacheConfiguration, DatabaseCachesConfiguration databaseCachesConfiguration) {
+        this.mtServiceCacheConfiguration = mtServiceCacheConfiguration;
+        this.databaseCachesConfiguration = databaseCachesConfiguration;
+    }
+
     @Bean
     @Override
     public CacheManager cacheManager() {
-        Cache defaultCache = new ConcurrentMapCache("default");
-        Cache localesCache = new ConcurrentMapCache("locales");
-        Cache pluralForm = new ConcurrentMapCache("pluralForms");
+        Cache defaultCache = new ConcurrentMapCache(CacheType.Names.DEFAULT);
+        Cache localesCache = new ConcurrentMapCache(CacheType.Names.LOCALES);
+        Cache pluralForm = new ConcurrentMapCache(CacheType.Names.PLURAL_FORMS);
+
+        TieredCache machineTranslationTieredCache = getMachineTranslationCache();
+
+        List<DatabaseCache> databaseCaches = getRuntimeDatabaseCaches();
```

Comment:
oh, you implemented both solutions (specific cache init for MT and generic case init)? I was thinking of an "or" in my previous comment, but I guess that works! The generic configuration won't be use right now, right?

Reply:
Removed generic case - not needed currently.